front/components/data_source/DocumentUploadOrEditModal.tsx

```diff
@@ -283,15 +327,15 @@ export const DocumentUploadOrEditModal = ({
         onClose(false);
       }}
       hasChanged={
-        !isDocumentError &&
-        !isDocumentLoading &&
-        !isContentLoading &&
-        !fileUploaderService.isProcessingFiles &&
-        isValidDocument
+        document
+          ? document.text !== documentState.text ||
+            documentState.name !== initialId
+          : documentState.name.trim().length > 0 ||
+            documentState.text.trim().length > 0
```

Comment:
nit: could we do it in the useEffect line 316 ?

Reply:
That would mean creating a new state variable for tracking whether the input has changed, and In practice the render rate and the useEffect update rates would be very close -> 1 per keystroke (because DocumentState would be a dep)